client/src/nl/sensorlab/videowall/animation/baseanimations/sorting/OptimizedBubbleSort.java

```diff
@@ -0,0 +1,20 @@
+package nl.sensorlab.videowall.animation.baseanimations.sorting;
+
+public class OptimizedBubbleSort extends BubbleSort {
+
+	private int c = 0;
+
+	public OptimizedBubbleSort(int[] arr) {
+		super(arr);
+		int c = 0;
+		num = 2;
+	}
+
+	public void sortStep() {
+		if (current + 1 == valuesArray.length-c) {
+			current = 0;
+			c++;
+		}
+		super.sortStep();
```

Comment:
Does calling the super method even work here, as sortStep is a abstract method? Conceptually this would lead to unwanted recursion